satpy/writers/cf_writer.py

```diff
@@ -495,20 +498,29 @@ def update_encoding(dataset, to_netcdf_kwargs):
     other_to_netcdf_kwargs = to_netcdf_kwargs.copy()
     encoding = other_to_netcdf_kwargs.pop('encoding', {}).copy()
 
-    _set_encoding_dataset_names(encoding, dataset)
+    _set_encoding_dataset_names(encoding, dataset, numeric_name_prefix)
     _set_default_chunks(encoding, dataset)
     _set_default_fill_value(encoding, dataset)
     _set_default_time_encoding(encoding, dataset)
 
     return encoding, other_to_netcdf_kwargs
 
 
+def _handle_dataarray_name(name, numeric_name_prefix):
+    if name[0].isdigit():
+        if numeric_name_prefix:
+            name = numeric_name_prefix + "_" + name
```

Comment:
I would rather leave full freedom to the user: let the user include the `_` themselves. So that giving `CHANNEL` as a prefix would result in `CHANNEL1` for example (which some users probably prefer)